pkg/model/api/v1/dashboard.go

```diff
@@ -212,24 +214,21 @@ func (d *Dashboard) verifyAndSetJSONReferences() error {
 	return nil
 }
 
-func (d *Dashboard) checkAndSetRef(ref *common.JSONRef) error {
-	// ref.Path should like that [ "spec", "panels", <name> ].
-	// So if the array is not equal to three then the reference is wrong.
-	if len(ref.Path) != 3 {
+func checkAndSetRef(ref *common.JSONRef, panels map[string]*Panel) error {
```

Comment:
This refactor seems interesting but is it tested somewhere? I'm afraid of any regression here as you don´t have all the same error messages here for example.

Reply:
Maybe this refactor could take place in another PR?

Reply:
Yes it's tested in both dashboard_test and ephemeraldashboard_test, it checks if the ref has been successfully resolved with `Object: panel` like here https://github.com/perses/perses/pull/1730/files#diff-0ad2f6727162626eb0f59d9f583d2520b8a5229a7a8a7b8cc1c390a447ce02b9R496